### ISSUE TITLE
`juju remove-application` doesn't fully cleanup `metacontroller-operator` resources

### ISSUE THREAD

**Issue by agileshaw** [canonical/metacontroller-operator#5]
#### Description
`juju remove-application` cannot fully remove the deployed `metacontroller-operator-charm` resources (statefulset and its running pod). 

#### Steps to reproduce the bug
1. From a fresh microk8s (after enabling storage, dns, and rbac), run `juju bootstrap microk8s micro`.
2. Create a new model `juju add-model test`.
3. Deploy `metacontroller-operator` charm by running `juju deploy metacontroller-operator --trust`. 
5. After being successfully deployed, remove the deployed charm: `juju remove-application metacontroller-operator`. 
6. Check microk8s "test" namespace via `microk8s kubectl get all -n test`. The statefulset `metacontroller-operator-charm` and its running pod `metacontroller-operator-charm-0` did not get removed.

#### Possible cause
In `juju debug-log`, I observed the following error:
```log
unit-metacontroller-operator-0: 2022-03-07 16:44:21 ERROR juju.worker.uniter.operation hook "remove" (via hook dispatching script: dispatch) failed: exit status 1
unit-metacontroller-operator-0: 2022-03-07 16:44:21 INFO juju.worker.uniter awaiting error resolution for "remove" hook
unit-metacontroller-operator-0: 2022-03-07 16:44:31 INFO juju.worker.uniter awaiting error resolution for "remove" hook
unit-metacontroller-operator-0: 2022-03-07 16:44:31 ERROR unit.metacontroller-operator/0.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "./src/charm.py", line 320, in <module>
    main(MetacontrollerOperatorCharm)
  File "/var/lib/juju/agents/unit-metacontroller-operator-0/charm/venv/ops/main.py", line 406, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-metacontroller-operator-0/charm/venv/ops/main.py", line 140, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-metacontroller-operator-0/charm/venv/ops/framework.py", line 278, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-metacontroller-operator-0/charm/venv/ops/framework.py", line 722, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-metacontroller-operator-0/charm/venv/ops/framework.py", line 767, in _reemit
    custom_handler(event)
  File "./src/charm.py", line 139, in _remove
    raise NotImplementedError()
NotImplementedError
unit-metacontroller-operator-0: 2022-03-07 16:44:31 ERROR juju.worker.uniter.operation hook "remove" (via hook dispatching script: dispatch) failed: exit status 1
unit-metacontroller-operator-0: 2022-03-07 16:44:31 INFO juju.worker.uniter awaiting error resolution for "remove" hook
```
It seems that the problem lies in the remove hook script.

#### Charm version
`metacontroller-operator` in charm-hub, revision 2

**Comment by benhoyt** [comment]
We (the Juju team) are hitting this too. One of our integration tests deploys and tears down `kubeflow-lite` which includes this charm, and fails on teardown because of this issue. Maybe the remove hook can just be deleted or commented out until it's needed? In the meantime for this Juju integration we're just going to not worry about teardown failure.

**Comment by ca-scribner** [comment]
Closed by #12
